app/controllers/setup_controller.rb

```diff
@@ -0,0 +1,38 @@
+class SetupController < ApplicationController
+
+  def index
+    @repositories = load_repositories
+    @labels = LabelsCarrier.new(load_events)
+
+  end
+
+  def update_labels
+    @labels = LabelsCarrier.new(load_events)
+
+    content = load_labels_content(@labels)
+    render json: { html: content}
+  end
+
+  private
+
+  def load_labels_content(labels)
+    if labels.uniq_labels.nil?
+      I18n.t('no_labels')
+    else
+      render_to_string( partial: 'setup/labels' ,  collection: labels.label_status, as: :label,
+                        locals: { select_options: labels.labels_name_list })
+    end
+  end
+
+  def load_repositories
+    Github.new(oauth_token: session[:token]).repos.list.map(&:name)
+  end
+
+  def load_events
+    if params[:repository].present?
+      Event.where(repo: params[:repository])
```

Comment:
And add `created` params like [here](https://github.com/jetthoughts/gh_metrics/pull/4/files#diff-12435b1e4e29226967cae794fbffb44cR21)